crates/yttrium/src/clear_signing/resolver.rs

```diff
@@ -0,0 +1,312 @@
+const DESCRIPTOR_WETH9: &str = include_str!("assets/descriptors/weth9.json");
+const DESCRIPTOR_AGGREGATION_ROUTER_V4: &str =
+    include_str!("assets/descriptors/aggregation_router_v4.json");
+const DESCRIPTOR_1INCH_AGG_ROUTER_V3: &str =
+    include_str!("assets/descriptors/1inch/calldata-AggregationRouterV3.json");
+const DESCRIPTOR_1INCH_AGG_ROUTER_V4_ETH: &str = include_str!(
+    "assets/descriptors/1inch/calldata-AggregationRouterV4-eth.json"
+);
+const DESCRIPTOR_1INCH_AGG_ROUTER_V4: &str =
+    include_str!("assets/descriptors/1inch/calldata-AggregationRouterV4.json");
+const DESCRIPTOR_1INCH_AGG_ROUTER_V5: &str =
+    include_str!("assets/descriptors/1inch/calldata-AggregationRouterV5.json");
+const DESCRIPTOR_1INCH_AGG_ROUTER_V6: &str =
+    include_str!("assets/descriptors/1inch/calldata-AggregationRouterV6.json");
+const DESCRIPTOR_1INCH_AGG_ROUTER_V6_ZKSYNC: &str = include_str!(
+    "assets/descriptors/1inch/calldata-AggregationRouterV6-zksync.json"
+);
+const DESCRIPTOR_1INCH_NATIVE_ORDER_FACTORY: &str =
+    include_str!("assets/descriptors/1inch/calldata-NativeOrderFactory.json");
+const DESCRIPTOR_AAVE_LPV2: &str =
+    include_str!("assets/descriptors/aave/calldata-lpv2.json");
+const DESCRIPTOR_AAVE_LPV3: &str =
+    include_str!("assets/descriptors/aave/calldata-lpv3.json");
+const DESCRIPTOR_AAVE_WETH_GATEWAY_V3: &str =
+    include_str!("assets/descriptors/aave/calldata-WrappedTokenGatewayV3.json");
+const INCLUDE_COMMON_TEST_ROUTER: &str =
+    include_str!("assets/descriptors/common-test-router.json");
+const INCLUDE_1INCH_COMMON_V4: &str =
+    include_str!("assets/descriptors/1inch/common-AggregationRouterV4.json");
+const INCLUDE_1INCH_COMMON_V6: &str =
+    include_str!("assets/descriptors/1inch/common-AggregationRouterV6.json");
+
+const ABI_ERC20: &str = include_str!("assets/abis/erc20.json");
+const ABI_UNISWAP_V3_ROUTER_V1: &str =
+    include_str!("assets/abis/uniswap_v3_router_v1.json");
+const ABI_WETH9: &str = include_str!("assets/abis/weth9.json");
+
+static INDEX: OnceLock<IndexMap> = OnceLock::new();
+const TYPED_INDEX_JSON: &str = include_str!("assets/index_eip712.json");
+const TYPED_DESCRIPTOR_1INCH_LIMIT_ORDER: &str =
+    include_str!("assets/descriptors/1inch/eip712-1inch-limit-order.json");
+const TYPED_DESCRIPTOR_1INCH_AGG_ROUTER_V6: &str =
+    include_str!("assets/descriptors/1inch/eip712-AggregationRouterV6.json");
+type TypedIndexMap = HashMap<String, String>;
+static TYPED_INDEX: OnceLock<TypedIndexMap> = OnceLock::new();
+
+fn load_index() -> IndexMap {
+    serde_json::from_str(INDEX_JSON)
+        .expect("clear signing index JSON must be valid")
+}
+
+fn index() -> &'static IndexMap {
+    INDEX.get_or_init(load_index)
+}
+
+fn load_typed_index() -> TypedIndexMap {
+    serde_json::from_str(TYPED_INDEX_JSON)
+        .expect("clear signing typed index JSON must be valid")
+}
+
+fn typed_index() -> &'static TypedIndexMap {
+    TYPED_INDEX.get_or_init(load_typed_index)
+}
```

Comment:
For `get_or_init` I'd combine these functions and use like a closure. Keeping them separate just makes it harder to read the code